src/tests/btree_helpers/btree_test_helper.hpp

```diff
@@ -303,14 +314,32 @@ struct BtreeTestHelper : public testing::Test {
     void print_keys() const { m_bt->print_tree_keys(); }
 
     void compare_files(const std::string& before, const std::string& after) {
-        std::ifstream b(before);
-        std::ifstream a(after);
-        std::ostringstream ss_before, ss_after;
-        ss_before << b.rdbuf();
-        ss_after << a.rdbuf();
-        std::string s1 = ss_before.str();
-        std::string s2 = ss_after.str();
-        ASSERT_EQ(s1, s2) << "Mismatch in btree structure";
+        std::ifstream b(before, std::ifstream::ate);
+        std::ifstream a(after, std::ifstream::ate);
+        if (a.fail() || b.fail()) {
+            LOGINFO("Failed to open file");
```

Comment:
Too much memory and overhead comparing large files earlier. So read in chunks and compare. Optimize to compare size first.